examples/traffic_histories_to_observations.py

```diff
@@ -1,130 +1,330 @@
+        self._logger = logging.getLogger(self.__class__.__name__)
 
-        collected_data = {}
-        _record_data(smarts.elapsed_sim_time, obs, collected_data)
+        smarts_seed(seed)
+
+        self._output_dir = output_dir
+        if output_dir and not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+        self._smarts = None
+        self._create_missions()
+        self._create_agent_interface()
+
+    def _create_missions(self):
+        self._missions = dict()
+        orig_missions = self._scenario.discover_missions_of_traffic_histories()
+        for v_id, mission in orig_missions.items():
+            veh_goal = self._scenario._get_vehicle_goal(v_id)
+            # TODO: get prefixed vehicle_id from TrafficHistoryProvider
+            veh_id = f"history-vehicle-{v_id}"
+            self._missions[veh_id] = replace(
+                mission, goal=PositionalGoal(veh_goal, radius=3)
+            )
+
+    def _create_agent_interface(
+        self, img_meters: int = 64, img_pixels: int = 256, action_space="TargetPose"
+    ):
+        # In future, allow for explicit mapping of vehicle_ids to agent_ids.
+        done_criteria = DoneCriteria(
+            collision=True,
+            off_road=True,
+            off_route=False,
+            on_shoulder=False,
+            wrong_way=False,
+            not_moving=False,
+            agents_alive=None,
+        )
+        max_episode_steps = 800
+        road_waypoint_horizon = 50
+        waypoints_lookahead = 50
+        self.agent_interface = AgentInterface(
+            accelerometer=True,
+            action=ActionSpaceType[action_space],
+            done_criteria=done_criteria,
+            drivable_area_grid_map=DrivableAreaGridMap(
+                width=img_pixels,
+                height=img_pixels,
+                resolution=img_meters / img_pixels,
+            ),
+            lidar=True,
+            max_episode_steps=max_episode_steps,
+            neighborhood_vehicles=True,
+            ogm=OGM(
+                width=img_pixels,
+                height=img_pixels,
+                resolution=img_meters / img_pixels,
+            ),
+            rgb=RGB(
+                width=img_pixels,
+                height=img_pixels,
+                resolution=img_meters / img_pixels,
+            ),
+            road_waypoints=RoadWaypoints(horizon=road_waypoint_horizon),
+            waypoints=Waypoints(lookahead=waypoints_lookahead),
+        )
+
+    def collect(
+        self, vehicles_with_sensors: Optional[Sequence[int]], headless: bool = True
+    ):
+        """Generate Observations from the perspective of one or more
+        social/history vehicles within a SMARTS scenario.
+
+        Args:
+            vehicles_with_sensors (Sequence[int], optional):
+                A list of vehicle_ids within the scenario to which to attach
+                sensors and record Observations.  If not specified, this will default
```

Comment:
I think there is a question of if there should be a difference in how `None` is treated in comparison to `[]`.

Consistency should say that `None` should have this behaviour but `[]` should just be empty.

Reply:
In general I would agree, but I think in this case there isn't a meaningful distinction since the result will be the same in both cases (script terminating due to having no valid vehicles to record observations for).